etcdctl/ctlv3/command/global.go

```diff
@@ -91,21 +99,22 @@ func mustClientFromCmd(cmd *cobra.Command) *clientv3.Client {
 		clientv3.SetLogger(log.New(os.Stderr, "grpc: ", 0))
 	}
 
-	endpoints, err := cmd.Flags().GetStringSlice("endpoints")
+	endpoints, err := getEndpointsFromCmd(cmd)
 	if err != nil {
 		ExitWithError(ExitError, err)
 	}
 	dialTimeout := dialTimeoutFromCmd(cmd)
 	sec := secureCfgFromCmd(cmd)
 	auth := authCfgFromCmd(cmd)
+	discovery := discoveryCfgFromCmd(cmd)
 
 	initDisplayFromCmd(cmd)
 
-	return mustClient(endpoints, dialTimeout, sec, auth)
+	return mustClient(endpoints, dialTimeout, sec, auth, discovery)
```

Comment:
I don't think `mustClient` needs to change at all; `mustClientFromCmd` can get the endpoints from discovery and  then pass that into `mustClient`. `secureCfg` will need a `serverName` field.

Reply:
in the last commit , you mentioned that the serverName does not belong to struct secureCfg ,i am confused that you have opposite opinion on the location of serverName in secureCfg; but it is a better way to add the serverName to serverName, as i just want to paas the domain & insecure to func newClientCfg and give value to tlsinfo.ServerName.

Reply:
sorry, the context wasn't clear from what was there before; I thought the intent was to pass the srv domain info into secureCfg, instead of using it to lock down tls for srv records

Reply:
thx, i will modify it soon